fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/data/SavingsAccountTransactionDTOV2.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.portfolio.savings.data;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.Objects;
+import java.util.Optional;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.apache.fineract.organisation.monetary.data.CurrencyData;
+import org.apache.fineract.organisation.monetary.domain.ApplicationCurrency;
+import org.apache.fineract.portfolio.account.data.AccountTransferData;
+import org.apache.fineract.portfolio.account.domain.AccountTransferTransaction;
+import org.apache.fineract.portfolio.paymentdetail.data.PaymentDetailData;
+import org.apache.fineract.portfolio.paymentdetail.domain.PaymentDetail;
+import org.apache.fineract.portfolio.paymenttype.data.PaymentTypeData;
+import org.apache.fineract.portfolio.savings.domain.SavingsAccount;
+import org.apache.fineract.portfolio.savings.service.SavingsEnumerations;
+import org.apache.fineract.useradministration.domain.AppUser;
+
+@Getter
+@AllArgsConstructor
+public class SavingsAccountTransactionDTOV2 {
+
+    private final Long transactionId;
+    private final Integer transactionType;
+    private final LocalDate transactionDate;
+    private final BigDecimal transactionAmount;
+    private final Long releaseIdOfHoldAmountTransaction;
+    private final String reasonForBlock;
+    private final LocalDateTime createdDate;
+    private final AppUser appUser;
+    private final String note;
+    private final BigDecimal runningBalance;
+    private final boolean reversed;
+    private final boolean reversalTransaction;
+    private final Long originalTxnId;
+    private final Boolean lienTransaction;
+    private final boolean isManualTransaction;
+    private final AccountTransferTransaction fromSavingsTransaction;
+    private final AccountTransferTransaction toSavingsTransaction;
+    private final SavingsAccount savingsAccount;
+    private final PaymentDetail paymentDetail;
+    private final ApplicationCurrency currency;
+
+    public static final SavingsAccountTransactionData tosavingsAccountTransactionData(SavingsAccountTransactionDTOV2 dto) {
```

Comment:
Are you planning to create a unit test for this class? Seems like there's way too much logic here than it should be.